### ISSUE TITLE
Slash Command Syntax Redesign -- Tracker

### ISSUE THREAD

**Issue by Starlight220** [cwarwicker/discord-Writer-Bot#107]
Due to some Discord policy changes, we're migrating the bot to use Slash Commands (see [this wiki page](https://github.com/cwarwicker/discord-Writer-Bot/wiki/Slash-Commands)).
Slash Command semantics are slightly different than the current prefix-based command system, so some of the commands will likely benefit from rephrasing them. Due to the popularity of some of these commands, we've decided to open the new syntax design for suggestions!

This issue will be used to list syntax suggestions/requests and will be updated.

**Comment by Starlight220** [comment]
Suggestion:
Slash commands don't support aliases (by design), so we're going to have to cut the aliases in the bot. I suggest to at first register both aliases as separate (sub)commands, with the difference that the alias will also print a message of this sort:
>The command alias `/sprint for x in y` is deprecated and will be removed in the close future.
Please use `/sprint start length:x delay:y` instead.

Removing `sprint for` in favor of a parameterized `sprint start` is a separate suggestion.

Thoughts?

**Comment by DarkJediQueen** [comment]
That seems so...clunky and not as easy to type honestly and I thought that the : commands were removed because of the native use that Discord has already built-in. Around time and having to then fight with what discord pops up already with the time using that. 

How would you be implementing the sprint for ## at .## command. That's the only one my server uses and it so simple. The other looks to me like it's going to cause more issues than solves. It's one a few characters off being twice as long as the original command.

**Comment by Starlight220** [comment]
@DarkJediQueen, I totally agree about the popups: they're extremely clunky and interfering for those who know and quickly type the commands (ie power users), but they're a lifesaver for those who have no idea what the commands are.

The problem with commands like `/sprint for ## at/in ##` is that every parameter has to have a name that's entered. A single required parameter at the beginning isn't much of a problem, but with multiple (and optional) parameters it's very easy to not hit `tab` and have the value entered for a parameter be missed. To make matters worse, we can't have both a base `/sprint` command as well as a `/sprint wc` subcommand (Discord framework limitations).
I've implemented a slashy version of `/sprint for x in y`, and much of my experience is based on that. Specifically in that matter, the `...at` variant should be easier than the `...in` variant: `in` and `for` can't be used as parameter names because they're reserved keywords in Python (while `at` isn't reserved). It might be possible to work around using `**arg` expansion or postfixing, but I'm not sure how that will turn out. I can look into some solutions at some point, I don't know what I might find.

**Comment by enigmalea** [comment]
The Discord API has been updated to accept a single line of text string for slash commands. I’m not sure what you’re using to implement slash commands, but if it supports the most updated API it should allow `/sprint [text string]` where `[text string]` is `for x in y` or whatever various constructions needed.

**Comment by Starlight220** [comment]
That would be one way to make it work (and probably what we'll do if there are still a lot of commands not ported in April), but if we do that we lose all the advantages of Slash Commands: type safety, input validation, autocomplete, and the quick documentation that pops up on each Slash Command.
